dotnet/src/SemanticKernel.Skills/Skills.Web/WebFileDownloadSkill.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.SkillDefinition;
+
+namespace Microsoft.SemanticKernel.Skills.Web;
+
+/// <summary>
+/// Skill to download web files.
+/// </summary>
+public class WebFileDownloadSkill : IDisposable
+{
+    /// <summary>
+    /// Parameter names.
+    /// <see cref="ContextVariables"/>
+    /// </summary>
+    public static class Parameters
+    {
+        /// <summary>
+        /// Where to save file.
+        /// </summary>
+        public const string FilePath = "filePath";
+    }
+
+    private readonly ILogger _logger;
+    private readonly HttpClientHandler _httpClientHandler;
+    private readonly HttpClient _httpClient;
+
+    /// <summary>
+    /// Constructor for WebFileDownloadSkill.
+    /// </summary>
+    /// <param name="logger">Optional logger.</param>
+    public WebFileDownloadSkill(ILogger<WebFileDownloadSkill>? logger = null)
+    {
+        this._logger = logger ?? NullLogger<WebFileDownloadSkill>.Instance;
+        this._httpClientHandler = new() { CheckCertificateRevocationList = true };
+        this._httpClient = new HttpClient(this._httpClientHandler);
+    }
+
+    /// <summary>
+    /// Downloads a file to a local file path.
+    /// </summary>
+    /// <param name="source">URI of file to download</param>
+    /// <param name="context">Semantic Kernel context</param>
+    /// <returns>Task.</returns>
+    /// <exception cref="KeyNotFoundException">Thrown when the location where to download the file is not provided</exception>
+    [SKFunction("Downloads a file to local storage")]
+    [SKFunctionInput(Description = "URL of file to download")]
+    [SKFunctionContextParameter(Name = Parameters.FilePath, Description = "Path where to save file locally")]
+    public async Task DownloadToFileAsync(string source, SKContext context)
+    {
+        this._logger.LogDebug($"{nameof(DownloadToFileAsync)} got called");
```

Comment:
Please use composite string formatting ("{0}", var)

Reply:
Using composite formatting makes sense when we have variables.  This is just a constant.

In other words, the "field" will never be used to distinguish between different invocations of this log since they will always be the same.

Do you still want me to change this?

Reply:
Per offline discussion, no changes needed :)